pkg/controller/machine/controller.go

```diff
@@ -54,6 +61,12 @@ func (c *MachineControllerImpl) Init(arguments sharedinformers.ControllerInitArg
 	// Use the lister for indexing machines labels
 	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Machines().Lister()
 
+	if c.nodeName == "" {
+		c.nodeName = os.Getenv(NodeNameEnvVar)
+		if c.nodeName == "" {
+			glog.Infof("environment variable %v is not set, this controller will not protect against deleting its own machine", NodeNameEnvVar)
```

Comment:
Maybe a warning instead of info (since we generally do expect the env var to be set)?

Reply:
Yes, good call.